File: tests/TryExceptEmptySemi.pas
namespace Demo;

interface

uses System;

type
  TryExceptEmptySemiExample = public class
  public
    class method DoNothing();
    class method DoCatch();
  end;

implementation

class method TryExceptEmptySemiExample.DoNothing();
begin
  try
    Console.WriteLine('A');
  except;
  end;
end;

class method TryExceptEmptySemiExample.DoCatch();
begin
  try
    Console.WriteLine('A');
  except;
    Console.WriteLine('B');
  end;
end;

end.

File: tests/TryExceptEmptySemi.cs
using System;

namespace Demo {
    public partial class TryExceptEmptySemiExample {
        public static void DoNothing() {
            try
            {
                Console.WriteLine("A");
            }
            catch (Exception)
            {}
        }
        public static void DoHandle() {
            try
            {
                Console.WriteLine("B");
            }
            catch (Exception)
            {
                Console.WriteLine("Error");
            }
        }
    }
}
